flink-connector-aws/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/FanOutShardSubscriber.java

```diff
@@ -329,6 +336,40 @@ private boolean isInterrupted(final Throwable throwable) {
         return false;
     }
 
+    private boolean isDefinedAsRecoverable(Throwable cause) {
+        // non-customisable list of exceptions that should be recovered (retried indefinitely).
+        if (cause instanceof ReadTimeoutException || cause instanceof TimeoutException) {
+            // ReadTimeoutException occurs naturally under backpressure scenarios when full batches take longer to
+            // process than standard read timeout (default 30s). Recoverable exceptions are intended to be retried
+            // indefinitely to avoid system degradation under backpressure. The EFO connection (subscription) to Kinesis
+            // is closed, and reacquired once the queue of records has been processed.
+            return true;
+        }
+        return isConfiguredAsRecoverable(cause);
+    }
+
+    /**
+     * @param cause Throwable on which to base our exception search
+     * @return true if the input Throwable is configured as a Recoverable Error by the user
+     */
+    private boolean isConfiguredAsRecoverable(Throwable cause) {
+        if (this.recoverableErrorsConfig == null || this.recoverableErrorsConfig.hasNoConfig()) {
+            return false;
+        }
+        for (ExceptionConfig config : this.recoverableErrorsConfig.getExceptionConfigs()) {
+            String classPath = config.getExceptionClassPath();
+            try {
+                Class<Throwable> aClass = (Class<Throwable>) Class.forName(classPath);
+                Optional<Throwable> throwable = ExceptionUtils.findThrowable(cause, aClass);
+                return throwable.isPresent();
+            } catch (ClassNotFoundException e) {
+                // Class not found
+                return false;
+            }
```

Comment:
Not sure we should swallow this, since it implies incorrect configuration. Instead of parsing it here recommend we inflate in the connector initialisation and perform validation up front.

Reply:
Update: Looks like we already are doing this below. Do we really need this? If we pass in a `Class<?>` instead of `String` we can remove the inflation